### DISCUSSION TITLE
How do you actually end/close the tour guide? (is it a bug?)

### DISCUSSION THREAD

**Discussion by PKHDK** [hakimio/ngx-ui-tour#156] [Q&A] [answered]
So I initialized the whole tour with 3 steps and by the time I arrive at the third step I see the "End" button but when I click on it nothing happens. Same applies to the "X" on the top right corner.
I also tried calling `this.tourService.end();` and `this.tourService.end$.subscribe(() => this.tourService.end())` and I even manually checked the ID of the step via the stepShow$ observable.. But I just can't get the tour to close!

I'm using the newest Angular version (16.2.4) and the newest ngx-ui-tour-md-menu (11.0.5). I also tried it on my other app where I'm still using modules instead of standalone components (v16.0.0) but it's also not closing there.

I also checked out the example source code here https://github.com/hakimio/ngx-ui-tour/blob/master/src/app/md-menu/md-menu.component.ts but I see no differences except for the provider with the "provideTourDirective" but I should not need that when I already imported the TourMatMenuModule right? Oh and I copied over the whole custom template stuff from https://github.com/hakimio/ngx-ui-tour/blob/master/src/app/md-menu/examples/default-template.txt over to my code since I thought maybe just using `<tour-step-template></tour-step-template>` is not enough (even though the docs say so)

I read through the docs a thousand times now, am I really just stupid or is it a bug? How do you close this whole thing?
And while we are at it - how can I fire a custom click event when the "End" button is pressed? (I want to notify the backend that the tourguide has been read)

**Comment by hakimio** [comment]
- No, you don't need `provideTourDirective()`.
- No, you don't have to provide your own custom template

Can you modify the "StackBlitz" demo to reproduce your issue:
https://stackblitz.com/edit/angular-wau2z8

**Reply by hakimio** [reply]
Also, do you get any errors?
Does it close when you press `escape` keyboard key?

**Reply by PKHDK** [reply]
Ok, I reproduced it in your stackblitz and suddenly it worked?! https://stackblitz.com/edit/angular-wau2z8-wete1b?file=src%2Fapp%2Fui-tour-demo.component.ts

Yes, I can at least close the tour with the "escape" key but the backdrop stays! (dimming most of the site). Thanks for that tip.
No, I don't receive a single error - just nothing happens when clicking "End" or "X".

So, next I will just try to downgrade either the lib or my angular version or both to test if it will work then.
And the other day I tried to remove zone.js for the most part since the application was on v13 for quite a while (only have an import 'zone.js' in my main.ts file by now. But I doubt that it is zone.js fault.

**Reply by hakimio** [reply]
> Ok, I reproduced it in your stackblitz and suddenly it worked?!

So, can you reproduce your issue on stackblitz or not?

> And the other day I tried to remove zone.js

That's a bad idea.

**Reply by PKHDK** [reply]
Sorry, I meant I wrote my code the same way in stackblitz but it actually works there! (but not in my codebase) 
Edit: using your versions as you stated it, I did not touch the angular and lib version in stackblitz, I will try doing that next to see if that is the problem

Yeah, I just removed references in angular.json and such but I still import it in main.ts - also I tested the lib in another project (with standalone components) where I did not manipulate zone.js but it also does NOT work (in the sense that the "End" button does not do anything and that pressing escape closes the tour but does not remove the backdrop) in the other app... but in stackblitz it suddenly works. So my only guess for now is to rollback the lib

**Reply by hakimio** [reply]
Create a github repository with a simple reproduction of your issue.

**Reply by hakimio** [reply]
Did you manage to find out what the issue was?

**Reply by PKHDK** [reply]
Sorry, no - I moved on to [angular shepherd](https://github.com/shipshapecode/angular-shepherd) which seems to work fine (no image or video support though which I actually liked about this ngx lib). 
Unfortunately I'm working on company code so I can't upload the code on a repo - and duplicating it would require too much time.

But thanks for providing your help anyway! I hope it's only a very rare unfortunate combination of some libs or configurations that cause this weird 'not closing' bug.

**Reply by hakimio** [reply]
If you are able to reproduce the issue in a new Angular 16 project, I could try to fix it. Without reproduction there is not much I can do here.

**Comment by hakimio** [comment, marked as answer]
Fixed in `12.0.3`
